src/filter/json_encode_filter.rs

```diff
@@ -0,0 +1,139 @@
+//! Convert LogLine events into Raw events encoded as JSON.
+//!
+//! This filter takes LogLines and encodes them into JSON, emitting the encoded
+//! event as a Raw event. This allows further filters or sinks to operate on the
+//! JSON without needing to understand a LogLine event in particular.
+//! If the LogLine value is a valid JSON object and parse_line config option is true,
+//! then the JSON will be merged with LogLine metadata. Otherwise, the original line
+//! will be included simply as a string.
+
+use chrono::DateTime;
+use chrono::naive::NaiveDateTime;
+use chrono::offset::Utc;
+use filter;
+use metric;
+use metric::TagMap;
+use rand::random;
+use serde_json;
+use serde_json::Value;
+use serde_json::map::Map;
+use std::iter::FromIterator;
+use std::sync::Arc;
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+lazy_static! {
+    /// Total number of logline processed
+    pub static ref JSON_ENCODE_LOG_PROCESSED: Arc<AtomicUsize> = Arc::new(AtomicUsize::new(0));
+    /// Total number of logline with JSON value successfully parsed
+    pub static ref JSON_ENCODE_LOG_PARSED: Arc<AtomicUsize> = Arc::new(AtomicUsize::new(0));
```

Comment:
We're counting these things but you haven't added any code to publish them in the internal source. Check out `src/source/internal.rs`